cloudinit/distros/debian.py

```diff
@@ -300,11 +300,38 @@ def update_package_sources(self):
     def get_primary_arch(self):
         return util.get_dpkg_architecture()
 
-    def set_keymap(self, layout, model, variant, options):
-        # Let localectl take care of updating /etc/default/keyboard
-        distros.Distro.set_keymap(self, layout, model, variant, options)
-        # Workaround for localectl not applying new settings instantly
+    def set_keymap(self, layout: str, model: str, variant: str, options: str):
+        # localectl is broken on some versions of Debian. See
+        # https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/2030788 and
+        # https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1038762
+        #
+        # Instead, write the file directly. According to the keyboard(5) man
+        # page, this file is shared between both X and the console.
+
+        contents = "\n".join(
+            [
+                "# This file was generated by cloud-init",
+                "",
+                f'XKBMODEL="{model}"',
+                f'XKBLAYOUT="{layout}"',
+                f'XKBVARIANT="{variant}"',
+                f'XKBOPTIONS="{options}"',
+                "",
+                'BACKSPACE="guess"',  # This is provided on default installs
```

Comment:
Might be a nitpick... Is it possible to allow the user to configure this, but still default to "guess" incase they do not configure this option.

Reply:
While a good thought, the implementation is highly debian-specific given that this implementation is writing directly to /etc/default/keyboard. Surfacing such and option is something that won't be supportable easily on any other distribution that is calling `localectl` directly, because that command doesn't support setting backspace values.